engine/src/test/java/io/camunda/zeebe/engine/processing/processinstance/CreateProcessInstanceSupportedElementTest.java

```diff
@@ -0,0 +1,233 @@
+                .startEvent()
+                .serviceTask(
+                    START_ELEMENT_ID,
+                    t ->
+                        t.zeebeJobType(JOBTYPE)
+                            .multiInstance(m -> m.parallel().zeebeInputCollectionExpression("[1]")))
+                .endEvent()
+                .done(),
+            Collections.emptyMap()),
+        new Scenario(
+            BpmnElementType.CALL_ACTIVITY,
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .callActivity(START_ELEMENT_ID, c -> c.zeebeProcessId(CHILD_PROCESS_ID))
+                .endEvent()
+                .done(),
+            Collections.emptyMap()),
+        new Scenario(
+            BpmnElementType.BUSINESS_RULE_TASK,
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .businessRuleTask(START_ELEMENT_ID, b -> b.zeebeJobType(JOBTYPE))
+                .done(),
+            Collections.emptyMap()),
+        new Scenario(
+            BpmnElementType.SCRIPT_TASK,
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .scriptTask(START_ELEMENT_ID, b -> b.zeebeJobType(JOBTYPE))
+                .done(),
+            Collections.emptyMap()),
+        new Scenario(
+            BpmnElementType.SEND_TASK,
+            Bpmn.createExecutableProcess(PROCESS_ID)
+                .startEvent()
+                .sendTask(START_ELEMENT_ID, b -> b.zeebeJobType(JOBTYPE))
+                .done(),
+            Collections.emptyMap()));
+  }
+
+  @Test
+  public void testProcessInstanceCanStartAtElementType() {
+    // given
+    ENGINE.deployment().withXmlResource(scenario.modelInstance).deploy();
+    if (scenario.type == BpmnElementType.CALL_ACTIVITY) {
+      ENGINE.deployment().withXmlResource(getChildProcess()).deploy();
+    }
+
+    // when
+    final long instanceKey =
+        ENGINE
+            .processInstance()
+            .ofBpmnProcessId(PROCESS_ID)
+            .withStartInstruction(START_ELEMENT_ID)
+            .withVariables(scenario.variables)
+            .create();
+
+    // then
+    assertThat(
+            RecordingExporter.processInstanceRecords()
+                .withProcessInstanceKey(instanceKey)
+                .onlyEvents()
+                .limit(
+                    r ->
+                        r.getValue().getBpmnElementType() == scenario.type
+                            && r.getIntent() == ProcessInstanceIntent.ELEMENT_ACTIVATED))
+        .extracting(record -> record.getValue().getBpmnElementType(), Record::getIntent)
+        .containsSubsequence(
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_ACTIVATING),
+            tuple(BpmnElementType.PROCESS, ProcessInstanceIntent.ELEMENT_ACTIVATED),
+            tuple(scenario.type, ProcessInstanceIntent.ELEMENT_ACTIVATING),
+            tuple(scenario.type, ProcessInstanceIntent.ELEMENT_ACTIVATED));
```

Comment:
`containsSubsequence` 
> Verifies that the actual group contains the given subsequence in the correct order **(possibly with other values between them)**.

`containsSequence` 
>Verifies that the actual group contains the given sequence in the correct order and **without extra values between the sequence values**

🔧 We should probably assert that there are no records in between these records. For example, if there is a START_EVENT ELEMENT_ACTIVATED between the PROCESS ELEMENT_ACTIVATED and the scenario.type ELEMENT_ACTIVATING, then it seems the start anywhere didn't work correctly.

🔧 In addition, we could also add explicitly `doesNotContain` of START_EVENT ELEMENT_ACTIVATING/ACTIVATED

Reply:
I can change it to a sequence instead of a subsequence. The start event however creates the problem with (event)subprocesses and call activities, so I will leave this out. It's not really necessary imo anyways, as we verify these elements get activates immediately after the process is activated. That would already rule out the activation of a start event in between.